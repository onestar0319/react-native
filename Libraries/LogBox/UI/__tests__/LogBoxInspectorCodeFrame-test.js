/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

'use strict';

const React = require('react');
const LogBoxInspectorCodeFrame = require('../LogBoxInspectorCodeFrame').default;
const render = require('../../../../jest/renderer');

describe('LogBoxInspectorCodeFrame', () => {
  it('should render null for no code frame', () => {
    const output = render.shallowRender(
      <LogBoxInspectorCodeFrame codeFrame={null} />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render a code frame', () => {
    const output = render.shallowRender(
      <LogBoxInspectorCodeFrame
        codeFrame={{
          fileName: '/path/to/RKJSModules/Apps/CrashReact/CrashReactApp.js',
          location: {row: 199, column: 0},
          content: `  197 | });
    198 |
  > 199 | export default CrashReactApp;
        | ^
    200 |`,
        }}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render a code frame without a location', () => {
    const output = render.shallowRender(
      <LogBoxInspectorCodeFrame
        codeFrame={{
          fileName: '/path/to/RKJSModules/Apps/CrashReact/CrashReactApp.js',
          location: null,
          content: `  197 | });
    198 |
  > 199 | export default CrashReactApp;
        | ^
    200 |`,
        }}
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
