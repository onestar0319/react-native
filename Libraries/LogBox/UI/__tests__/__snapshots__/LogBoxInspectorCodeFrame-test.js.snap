// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorCodeFrame should render a code frame 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
      "marginTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(51, 51, 51, 1)",
        "borderRadius": 3,
        "marginLeft": 10,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <ScrollViewMock
      horizontal={true}
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "flex": 1,
            "fontFamily": "Menlo",
            "fontSize": 12,
            "includeFontPadding": false,
            "lineHeight": 20,
          }
        }
      >
          197 | });
    198 |
  &gt; 199 | export default CrashReactApp;
        | ^
    200 |
      </Text>
    </ScrollViewMock>
  </View>
  <Text
    style={
      Object {
        "color": "rgba(255, 255, 255, 0.5)",
        "flex": 1,
        "fontFamily": "Menlo",
        "fontSize": 12,
        "includeFontPadding": false,
        "lineHeight": 20,
        "textAlign": "center",
      }
    }
  >
    CrashReactApp.js
     
    (199:0)
  </Text>
</View>
`;

exports[`LogBoxInspectorCodeFrame should render null for no code frame 1`] = `null`;
